lib/ExponentialBackoff.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var ExponentialBackoff = function(minimum_backoff, exponent, starting_backoff) {
+    if (minimum_backoff === undefined) {
+        minimum_backoff = 50;
+    }
+
+    if (exponent === undefined) {
+        exponent = 2.0;
+    }
+
+    if (starting_backoff === undefined) {
+        starting_backoff = minimum_backoff;
+    }
+
+    var backoff = starting_backoff;
+
+    return {
+        getBackoff: function() {
```

Comment:
It's better to define the class as follows:

```javascript
// constructor
var ExponentialBackoff = function(...) { ... }

/**
* makes it easier to jsdoc as well (which we use)
 */
ExponentialBackoff.prototype.getBackoff = function getBackoff(...) { ... };

module.exports = ExponentialBackoff;
```

When you define everything inside the constructor, that code gets executed in it's entirety each time an instance of the class is instantiated. That can be slow.